base/strings/basic.jl

```diff
@@ -140,6 +140,32 @@ prevind(s::AbstractArray   , i::Integer) = i-1
 nextind(s::DirectIndexString, i::Integer) = i+1
 nextind(s::AbstractArray   , i::Integer) = i+1
 
+function prevind(s::String, i::Integer)
+    data = s.data
+    stop = endof(s.data)
```

Comment:
Why not call `stop` `e` as in the `AbstractString` code? Also, since `String` is immutable, you don't actually need to store `data` in a variable for performance.